crates/matrix-sdk/src/latest_events/mod.rs

```diff
@@ -536,10 +536,30 @@ impl RoomLatestEvents {
     /// Update the latest events for the room and its threads, based on the
     /// send queue update.
     async fn update_with_send_queue(&mut self, send_queue_update: &RoomSendQueueUpdate) {
-        self.for_the_room.update_with_send_queue(send_queue_update).await;
+        // Get the power levels of the user for the current room if the `WeakRoom` is
+        // still valid.
+        //
+        // Get it once for all the updates of all the latest events for this room (be
+        // the room and its threads).
+        let room = self.weak_room.get();
+        let power_levels = match &room {
```

Comment:
(not technically a big deal, but this is a lie, as it's `user_id_and_power_levels`, right?

Reply:
Yes! I wanted to keep it short this time, because it's the user ID associated to the room power levels… I can rename if you prefer?